### PR TITLE
MT3-239 #ready-for-test #comment Add tag component

### PR DESCRIPTION
**Pull request by CJWaszczuk**
# What?

Adds a tag component

# Why?

Users may want to be able to use a tag to indicate the phase of their product, or to display the current state of a process. 

# References 

`lbh-frontend` library: [Github](https://github.com/LBHackney-IT/LBH-frontend/tree/master/src/lbh/components/lbh-tag) | [Style guide](https://lbh-frontend.herokuapp.com/components/lbh-tag)
`govuk-frontend` library : [Github](https://github.com/alphagov/govuk-frontend/tree/master/src/govuk/components/tag) | [Style guide](https://design-system.service.gov.uk/components/tag/)

### CODE REVIEW

File: src/components/Tag.test.tsx
Comment (erbridge) [inline src/components/Tag.test.tsx]:
```suggestion
import { create } from "react-test-renderer";
```

In general, let's do this. Don't feel like you need to change this one unless you are in there anyway.

File: src/components/Tag.tsx
Comment (erbridge) [inline src/components/Tag.tsx]:
```suggestion
 * A component for indicating the status of something, such as an item on a task list or a phase banner.
```

The design system's explanation is good. Let's reuse it if we can.

**Review by erbridge — COMMENTED** [review]
Minor doc suggestion. Otherwise, looks good.

**Comment by erbridge** [comment]
Make sure to rebase this and update to match the current style.

File: docs/api/assets/css/main.css
Comment (erbridge) [inline docs/api/assets/css/main.css]: Interesting. This is because the theme changed. I'll look into getting the tests to fail if the docs have changed but not been committed, so we can avoid this.

File: src/components/Tag.tsx
Comment (erbridge) [inline src/components/Tag.tsx]: Just to confirm: `lbh-frontend` has no tag specific styles, right?

File: src/components/Tag.tsx
Comment (erbridge) [inline src/components/Tag.tsx]: Pull the props destructuring into the body of the funtcion and give the argument a type, so the docs can document it (and regenerate the docs).

File: src/components/Tag.tsx
Comment (CJWaszczuk) [reply src/components/Tag.tsx]: The tag specific styles for `lbh-frontend` are defined via phase-banner. The tag used on its own has no lbh styles, probably because there was no reason to do so yet, but I feel it may be something that appears when we get to [task lists](https://design-system.service.gov.uk/patterns/task-list-pages/) (verified / done stamps)  . Might be worth raising this to @LBHELewis  because at the moment, a tag called outside the phase banner will look different stylistically to a tag called within it.

Comment (erbridge) [reply src/components/Tag.tsx]: This is related to LBHackney-IT/LBH-frontent#35.

**Comment by LBHELewis** [comment]
Once https://github.com/LBHackney-IT/LBH-frontend/pull/41 is merged, and the references to `govuk-tag` are replaced with `lbh-tag` this is looking good to go.

**Comment by erbridge** [comment]
Let's hold off on merging this until LBHackney-IT/LBH-frontend#41 is resolved.

**Comment by CJWaszczuk** [comment]
Made the changes to tag with regards to styling, would be good to have a final check over this before merge.

File: src/components/Tag.tsx
Comment (erbridge) [inline src/components/Tag.tsx]:
```suggestion
    <strong id={id} className={classnames("lbh-tag", "govuk-tag", className)}>
```

I think?

**Review by erbridge — APPROVED** [review]
LGTM

**Pull request merged**
